README.md

```diff
@@ -1,4 +1,253 @@
+
+In one of your [django app configs](https://docs.djangoproject.com/en/dev/ref/applications/#for-application-authors):
+
+```python
+def add_csv_permissions_watcher(sender, **kwargs):
+    """In dev we want to reload if the csv permission file changes"""
+    sender.extra_files.add(settings.CSV_PERMISSIONS_PATH)
+
+class MySiteAppConfig(AppConfig):
+    name = "my_site"
+    verbose_name = "My Site"
+
+    def ready(self):
+        if settings.DEBUG:
+            from django.utils.autoreload import autoreload_started
+
+            autoreload_started.connect(add_csv_permissions_watcher)
+```
+
+If you're using [`runserver_plus`](https://django-extensions.readthedocs.io/en/latest/runserver_plus.html)
+from `django-extensions` you can add your CSV files to
+[`settings.RUNSERVER_PLUS_EXTRA_FILES`](https://django-extensions.readthedocs.io/en/latest/runserver_plus.html#configuration).   
+
+  
+### The CSV File
+
+An example permission file:
+
+```csv
+Model,     App,           Action,            Is Global, admin, assistant, customer
+
+# Comment lines and blank lines will be ignored
+
+Publisher, library,       add,               yes,       yes,
+Publisher, library,       view,              no,        all,
+Publisher, library,       change,            no,        all,
+Publisher, library,       delete,            no,        all,
+
+Book,      library,       add,               yes,       yes,   yes,
+Book,      library,       view,              no,        all,   all,
+Book,      library,       change,            no,        all,   all,
+Book,      library,       delete,            no,        all,   all,
+
+Loan,      library,       add,               yes,       yes,   yes,       yes,
+Loan,      library,       view,              no,        all,   all,       own,
+Loan,      library,       change,            no,        all,   all,
+Loan,      library,       delete,            no,        all,   all,
+
+# The model column can be blank:
+
+,          library,       report_outstanding,yes,      yes,   yes,
+,          library,       report_popularity, yes,      yes,   yes,
+```
+
+The first 4 columns define the permission details:
+
+**Model** is used to resolve the permission name but is otherwise not used. There is no checks that objects passed to the `has_perm()` actually match the correct type.
+
+**App** is used to resolve the permission name and model.
+
+**Action** is an arbitrary identifier that is used to resolve the permission name.
+
+**Is Global** whether the permission is global or per-object (see "Global Permission" section below)
+
+**Evaluators**
+
+The next columns define permission "evaluators".
+
+Built-in evaluators are:
+
+* `all` - user has permission for all objects. Will raise an error  if an object is not passed to `has_perm()`
+* `yes` - user has permission globally. Will raise an error if an object is passed to `has_perm()`.
+* (empty cell) -- user does not have permission (global or per-object) 
```

Comment:
so the `custom:` prefix is no longer required for custom rules?

Reply:
Depends on what you want to do -- it's left entirely up to the application.

`csv_permissions.legacy.resolve_custom_evaluator` can be used to get the old behaviour, but the intention is that other than the core yes/all/'' evaluators you'll define them in your application.

I anticipate that at some point we'll have a built-in `rule:foo` that will check that `foo` is a django-rules predicate but didn't want to do this until I'd used it on a project to make sure that the interface is settled (eg, what do we do about the same rule name in 2 different apps? we can fully qualify it but that's going to make the cells contents quite long -- do we want to allow unqualified rules if they're unambiguous? perhaps is better to provide a django-rules evaluator in csv_permissions but not make it default, and let the app to decide what to do)